### PR TITLE
feat: Switch UiAutomator2 server to W3C protocol

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This change is breaking and won't allow to init the server using JSONWP caps anymore.

Affected areas:
- Session capabilities 
- Element key in responses
- Errors in responses

### CODE REVIEW

File: app/src/test/java/io/appium/uiautomator2/utils/w3c/W3CCapsUtilsTests.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class W3CCapsUtilsTests {
```
Comment (KazuCocoa) [inline app/src/test/java/io/appium/uiautomator2/utils/w3c/W3CCapsUtilsTests.java]: How about adding a test case which has non-appium prefix caps like `"{\"goog:chromeOptions\":{\"args\":[\"--disable-popup-blocking\"]}}"`? It should be valid.

Comment (mykola-mokhnach) [reply app/src/test/java/io/appium/uiautomator2/utils/w3c/W3CCapsUtilsTests.java]: yep, will do. In general, such caps should stay unchanged

File: app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java [hunk 2]
```diff
@@ -57,4 +64,44 @@ public static Integer readInteger(JSONObject payload, String name, boolean isReq
         }
         return null;
     }
+
+    public static Map<String, Object> toMap(JSONObject obj) {
```
Comment (KazuCocoa) [inline app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java]: Nullable annotation?

File: app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java [hunk 3]
```diff
@@ -57,4 +64,44 @@ public static Integer readInteger(JSONObject payload, String name, boolean isReq
...
+            } else if (value instanceof JSONArray) {
+                result.put(key, toList((JSONArray) value));
+            } else {
+                result.put(key, value);
+            }
+        }
+        return result;
+    }
+
+    public static List<Object> toList(JSONArray obj) {
```
Comment (KazuCocoa) [inline app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java]: Nullable?

File: app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java (hunk 2 above)
Comment (mykola-mokhnach) [reply app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java]: the result will never be null. It can be an empty map in the worst case. Same below

Comment (KazuCocoa) [reply app/src/main/java/io/appium/uiautomator2/utils/JSONUtils.java]: Ah, okay. I thought null since they returned null when obj == null

**Pull request merged**